cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -920,12 +920,22 @@ def _validate_obsm(self):
         obsm_with_x_prefix = 0
         for key, value in self.adata.obsm.items():
             issue_list = self.errors
+
+            if not key[0].isalpha():
```

Comment:
The actual Seurat v5 regex is matching against `^[a-zA-Z][a-zA-Z0-9]*$` for keys (or suffixes of keys that begin with `X_`). Which also encompasses making sure the key is alphanumeric overall. If that isn't already captured elsewhere, we might want to just check against this regex.

Reply:
That was not captured in the original Data Viz requirements which only prohibited white space characters. 

I was writing an update for Nik this morning:

- {suffix} MUST be at least one character in length.
- The first character of {suffix} MUST be a letter of the alphabet.
- {suffix} MUST NOT contain whitespace characters.

I've been trying hard to avoid ABNF in the schema, but perhaps it's time to consider using RegEx expressions for clarity; otherwise:

- {suffix} MUST be at least one character in length.
- The first character of {suffix} MUST be a letter of the alphabet.
- The remaining characters of {suffix} MUST be alphanumeric characters.

Reply:
It does seem simpler to just write:

{suffix} MUST match the regular expression pattern "^[a-zA-Z][a-zA-Z0-9]*$".

Reply:
@nayib-jose-gloria Good point - I'll just have the code use the regex pattern.

@brianraymor For clarity including the regular expression pattern is definitely useful. But I think it would be better to include the regex in addition to the English, rather than replacing it, since most people can't read regex and immediately understand what it means.